[SQL] Date operations mentioned in the documentation fail with CalciteContextException

The following operations are mentioned in the documentation: 
- DATE + INTEGER
- DATE - INTEGER
- DATE + TIME
- DATE - DATE

However, performing these operations fail as they are potentially not supported in Calcite? 

Consider a test as follows: 
```sql
                SELECT DATE '2024-10-11' + TIME '20:10:10';
                 date
                ----------
                 2024-10-11 20:10:10
                (1 row)
```

This fails with:
```java
(no input file): Error in SQL statement
(no input file):1:26: error: Error in SQL statement: Cannot apply '+' to arguments of type '<DATE> + <TIME(0)>'. Supported form(s): '<NUMERIC> + <NUMERIC>'
'<DATETIME_INTERVAL> + <DATETIME_INTERVAL>'
'<DATETIME> + <DATETIME_INTERVAL>'
'<DATETIME_INTERVAL> + <DATETIME>'
    1|CREATE VIEW VV AS SELECT DATE '2024-10-11' + TIME '20:10:10'
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


java.lang.RuntimeException: org.apache.calcite.runtime.CalciteContextException: From line 1, column 26 to line 1, column 60: Cannot apply '+' to arguments of type '<DATE> + <TIME(0)>'. Supported form(s): '<NUMERIC> + <NUMERIC>'
'<DATETIME_INTERVAL> + <DATETIME_INTERVAL>'
'<DATETIME> + <DATETIME_INTERVAL>'
'<DATETIME_INTERVAL> + <DATETIME>'

	at org.dbsp.sqlCompiler.compiler.DBSPCompiler.compileInternal(DBSPCompiler.java:298)
	at org.dbsp.sqlCompiler.compiler.DBSPCompiler.compileStatement(DBSPCompiler.java:331)
	at org.dbsp.sqlCompiler.compiler.DBSPCompiler.compileStatement(DBSPCompiler.java:351)
	at org.dbsp.sqlCompiler.compiler.sql.SqlIoTest.compare(SqlIoTest.java:145)
	at org.dbsp.sqlCompiler.compiler.sql.SqlIoTest.q(SqlIoTest.java:190)
	at org.dbsp.sqlCompiler.compiler.sql.SqlIoTest.qs(SqlIoTest.java:277)
	at org.dbsp.sqlCompiler.compiler.sql.SqlIoTest.qs(SqlIoTest.java:286)
	at org.dbsp.sqlCompiler.compiler.sql.simple.TimeArithmeticTests.testDateAddInteger(TimeArithmeticTests.java:322)
        .
        .
        .
Caused by: org.apache.calcite.runtime.CalciteContextException: From line 1, column 26 to line 1, column 60: Cannot apply '+' to arguments of type '<DATE> + <TIME(0)>'. Supported form(s): '<NUMERIC> + <NUMERIC>'
'<DATETIME_INTERVAL> + <DATETIME_INTERVAL>'
'<DATETIME> + <DATETIME_INTERVAL>'
'<DATETIME_INTERVAL> + <DATETIME>'

```

--- a/docs/sql/datetime.md
+++ b/docs/sql/datetime.md
@@ -231,15 +231,11 @@ The following arithmetic operations are supported:
 
 | Operation                   | Result Type        | Explanation                                                      |
 |-----------------------------|--------------------|------------------------------------------------------------------|
-| `DATE` + `INTEGER`          | `DATE`             | Add a number of days to a date                                   |
 | `DATE` + `INTERVAL`         | `TIMESTAMP`        | Add an interval to a date                                        |
-| `DATE` + `TIME`             | `TIMESTAMP`        | Create a timestamp from parts                                    |
 | `INTERVAL` + `INTERVAL`     | `INTERVAL`         | Add two intervals; both must have the same type                  |
 | `TIMESTAMP` + `INTERVAL`    | `TIMESTAMP`        | Add an interval to a timestamp                                   |
 | `TIME` + `INTERVAL`         | `TIME`             | Add an interval to a time. Performs wrapping addition.           |
 | `-` `INTERVAL`              | `INTERVAL`         | Negate an interval                                               |
-| `DATE` - `DATE`             | `INTERVAL`         | Compute the interval between two dates                           |
-| `DATE` - `INTEGER`          | `DATE`             | Subtract a number of days from a date                            |
 | `DATE` - `INTERVAL`         | `DATE`             | Subtract an interval from a date                                 |
 | `TIME` - `TIME`             | `INTERVAL` (short) | Compute the difference between two times                         |
 | `TIME` - `INTERVAL`         | `TIME`             | Subtract an interval from a time. Performs wrapping subtraction. |

